clean up new user registration page a bit

s-cheng opened the pull request:
This PR is just cleaning up the styling a bit on the new user registration page (the one where you pick your username). This should basically not affect LW.

Old, desktop:
<img width="1440" alt="Screen Shot 2023-08-08 at 10 31 57 AM" src="https://github.com/ForumMagnum/ForumMagnum/assets/9057804/ddd9aa6f-55db-4e90-926e-1ddfcabedcdf">

New, desktop:
<img width="1440" alt="Screen Shot 2023-08-08 at 10 52 07 AM" src="https://github.com/ForumMagnum/ForumMagnum/assets/9057804/80789bfa-9c6a-49b5-8f62-51eabc9b4481">

-----
Old, mobile:
<img width="366" alt="Screen Shot 2023-08-08 at 10 54 46 AM" src="https://github.com/ForumMagnum/ForumMagnum/assets/9057804/7d21092c-8325-4a32-868c-0050e7556027">

New, mobile:
<img width="366" alt="Screen Shot 2023-08-08 at 10 59 48 AM" src="https://github.com/ForumMagnum/ForumMagnum/assets/9057804/8c6f957e-b802-4731-bb01-3ed70e6c3e2e">

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1205234270505745) by [Unito](https://www.unito.io)

s-cheng commented:
@agnesstenlund FYI I just cleaned up this page a bit, since I was registering a new user and didn't like how the text looked here - feel free to make more adjustments :)

jpaddison3 commented on packages/lesswrong/components/users/NewUserCompleteProfile.tsx:
```diff
@@ -28,6 +35,10 @@ const styles = (theme: ThemeType): JssStyles => ({
       color: theme.palette.grey[600],
     },
   },
+  sectionHeadingText: isEAForum ? {
```
I don't think less wrong uses this page at all? So shouldn't be necessary — just make it good for anyone using it, IMO.

s-cheng replied:
Oh interesting, I didn't know that, let me check with them

jpaddison3 reviewed (approved):
ASTLAC

s-cheng commented:
Sounds like LW doesn't use this, so I removed a bunch of forum-gating. Also I noticed that this page can have different backgrounds depending on which URL you hit (maybe that shouldn't happen but don't want to fix it now), so I updated it to account for the grey background version:

<img width="1603" alt="Screen Shot 2023-08-08 at 4 58 36 PM" src="https://github.com/ForumMagnum/ForumMagnum/assets/9057804/5aab4d8d-a63b-406f-b8db-f23f12b36e43">

s-cheng commented on packages/lesswrong/components/users/NewUserCompleteProfile.tsx:
```diff
@@ -146,7 +161,9 @@ const NewUserCompleteProfile: React.FC<NewUserCompleteProfileProps> = ({ current
       
       {/* Facebook user with no email fix (very small % of users) */}
```
@jpaddison3 Does this Facebook case apply to us? If not, then could we remove this code?
